### PR TITLE
DOCSP-45309: CMake option changes

### PR DESCRIPTION
**Pull request by norareidy**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-45309
Staging:
https://deploy-preview-75--docs-cpp.netlify.app/whats-new/#what-s-new-in-4.0
https://deploy-preview-75--docs-cpp.netlify.app/upgrade/#version-4.0

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [ ] Are all the links working?
- [ ] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/whats-new.txt [hunk 1]
```diff
@@ -93,6 +93,14 @@ What's New in 4.0
      - ``BSONCXX_POLY_USE_SYSTEM_MNMLSTC``.
      - ``BSONCXX_POLY_USE_BOOST``.
 
+   - Drops support for the ``MONGOCXX_OVERRIDE_DEFAULT_INSTALL_PREFIX`` configuration
+     option. This option is implicitly set to ``OFF``.
+
+   - Sets the ``ENABLE_TESTS`` configuration option to ``OFF`` by default. To
```
Comment (norareidy) [inline source/whats-new.txt:99]: Question for tech reviewer: is this a breaking change?

Comment (eramongodb) [reply source/whats-new.txt]: This is a build system breaking change which does not affect the API or ABI of the driver libraries. This only affects packaging recipes and/or build scripts which expect test targets to be enabled (most are likely already explicitly setting `ENABLE_TESTS=OFF`). It does not impact users of the driver libraries themselves.

File: source/upgrade.txt [hunk 2]
```diff
@@ -66,20 +66,97 @@ and upgrade versions.
...
+.. _upcoming-breaking-changes:
 .. _version-4.0-breaking-changes:
 
 Version 4.0 Breaking Changes
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-This driver version introduces the following breaking changes:
+.. _version-4.0-build-system-breaking:
+
+.. warning:: Build System Breaking Changes
```
Comment (eramongodb) [inline source/upgrade.txt:123]: Given this page is now entirely dedicated to documenting breaking changes, making every section a WARNING is probably overkill. Recommend turning these into subsections instead (as is currently done for [PyMongo](https://www.mongodb.com/docs/languages/python/pymongo-driver/current/upgrade/)):

> ## Breaking Changes
> > NOTE
> > The C++ driver follows semantic versioning. ...
>
> A breaking change is...
> ...
> ### Version 4.0 Breaking Changes
> #### Build System
> The v4.0 driver ... to the build system:
> - Drops support for Boost and MNMLSTC/core...
> ...
> #### API
> The v4.0 driver ... to the API:
> - Drops support for configuring...
> ...
> #### ABI
> The v4.0 driver ... to the ABI:
> - Drops support for ...
> ...
> ### Version 3.11 Breaking Changes
> ...

File: source/upgrade.txt
Comment (eramongodb) [inline source/upgrade.txt:1]: The "Upgrading from Version 3.0" example can be updated to refer to the 4.0 release instead:

> EXAMPLE
> **Upgrading to Version 4.0**
> If you're upgrading the driver from version 3.X to version 4.0, address all breaking changes which are listed for version 3.X up to and including version 4.0.

File: source/upgrade.txt [hunk 3]
```diff
@@ -66,20 +66,97 @@ and upgrade versions.
...
+
+     - Removes the ``ENABLE_FOO`` configuration variable.
+     - Changes the default value of the ``ENABLE_BAR`` configuration variable from
+       ``OFF`` to ``ON``.
+     - Drops support for platform A, compiler B, or architecture C.
+
+.. _upcoming-breaking-changes:
 .. _version-4.0-breaking-changes:
 
 Version 4.0 Breaking Changes
```
Comment (eramongodb) [inline source/upgrade.txt:118]: Recommend making these sections (version) visible under the "On this page" table of contents.

> On this page
> - Overview
> - Breaking Changes
>     - Version 4.0 Breaking Changes
>     - Version 3.11 Breaking Changes
>     - ...

Aside: perhaps the section names can be simplified to just "Version 4.0", etc.? See other comment regarding (sub)section layout.

File: source/upgrade.txt [hunk 4]
```diff
@@ -66,20 +66,97 @@ and upgrade versions.
   If you're upgrading the {+driver-short+} from version 3.0 to version 3.10,
   address all breaking changes listed for versions 3.1 to 3.10, if any.
 
+Types of Breaking Changes
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
```
Comment (eramongodb) [inline source/upgrade.txt:71]:
```suggestion
```
Suggest merging this section into a continuation of the "Breaking Changes" section and/or consider making both subsections of the "Overview" section such that the ToC looks something like:

> On this page
> - Overview
>     - "This page describes..."
>     - "This page describes..."
> - Breaking Changes
>     - "A breaking change is..."
>     - "New versions of the C++ driver can introduce..."
>     - Version 4.0
>         - ...
>     - Version 3.11
>     - ...

or (my preference):

> On this page
> - Overview
>     - "This page describes..."
>     - "A breaking change is..."
>     - "New versions of the C++ driver can introduce..."
> - Breaking Changes
>     - Version 4.0
>         - ...
>     - Version 3.11
>     - ...

File: source/get-started/download-and-install.txt [hunk 5]
```diff
@@ -59,7 +59,6 @@ Download and Install
                 cmake ..                                \
                     -DCMAKE_BUILD_TYPE=Release          \
                     -DCMAKE_CXX_STANDARD=17                     \
-                    -DMONGOCXX_OVERRIDE_DEFAULT_INSTALL_PREFIX=OFF
```
Comment (eramongodb) [inline source/get-started/download-and-install.txt:62]: Also remove the trailing backslash in the preceeding line, or consider folding the command into a single line.

**Review by eramongodb — CHANGES_REQUESTED** [review]

**Comment by eramongodb** [comment]
It looks like there is overlap between changes in this PR and https://github.com/mongodb/docs-cpp/pull/80. Please defer my earlier feedback concerning content layout organization to https://github.com/mongodb/docs-cpp/pull/80.

File: source/upgrade.txt [hunk 6]
```diff
@@ -141,6 +141,14 @@ The v4.0 driver introduces the following breaking changes to the build system:
   - ``BSONCXX_POLY_USE_SYSTEM_MNMLSTC``.
   - ``BSONCXX_POLY_USE_BOOST``.
 
+- Drops support for the ``MONGOCXX_OVERRIDE_DEFAULT_INSTALL_PREFIX`` configuration
+  option. This option is implicitly set to ``OFF``.
```
Comment (eramongodb) [inline source/upgrade.txt:145]:
```suggestion
- Drops support for the ``MONGOCXX_OVERRIDE_DEFAULT_INSTALL_PREFIX`` configuration
  option. Default CMake behavior for `CMAKE_INSTALL_PREFIX` is now respected.
```
Wording tweak to direct users to CMake options and its (default) behavior. I am having trouble suggesting an alternative wording, but the point we want to make is, "we no longer have any potentially-surprising behavior that differs from normal, common CMake behavior concerning the handling of the install prefix path".

**Review by eramongodb — APPROVED** [review]
Minor suggestion; otherwise, LGTM.

**Pull request merged**